asv_bench/benchmarks/pandas/frame_ctor.py

```diff
@@ -0,0 +1,225 @@
+import numpy as np
+
+import modin.pandas as pd
+from modin.pandas import (
+    NA,
+    Categorical,
+    DataFrame,
+    Float64Dtype,
+    MultiIndex,
+    Series,
+    Timestamp,
+    date_range,
+)
+
+import pandas._testing as tm
+
+try:
+    from pandas.tseries.offsets import (
+        Hour,
+        Nano,
+    )
+except ImportError:
+    # For compatibility with older versions
+    from pandas.core.datetools import (
+        Hour,
+        Nano,
+    )
+
+
+class FromDicts:
+    def setup(self):
+        N, K = 5000, 50
+        self.index = tm.makeStringIndex(N)
+        self.columns = tm.makeStringIndex(K)
+        frame = DataFrame(np.random.randn(N, K), index=self.index, columns=self.columns)
+        self.data = frame.to_dict()
+        self.dict_list = frame.to_dict(orient="records")
+        self.data2 = {i: {j: float(j) for j in range(100)} for i in range(2000)}
+
+        # arrays which we won't consolidate
+        self.dict_of_categoricals = {i: Categorical(np.arange(N)) for i in range(K)}
+
+    def time_list_of_dict(self):
+        DataFrame(self.dict_list)
+
+    def time_nested_dict(self):
+        DataFrame(self.data)
+
+    def time_nested_dict_index(self):
+        DataFrame(self.data, index=self.index)
+
+    def time_nested_dict_columns(self):
+        DataFrame(self.data, columns=self.columns)
+
+    def time_nested_dict_index_columns(self):
+        DataFrame(self.data, index=self.index, columns=self.columns)
+
+    def time_nested_dict_int64(self):
+        # nested dict, integer indexes, regression described in #621
+        DataFrame(self.data2)
+
+    def time_dict_of_categoricals(self):
+        # dict of arrays that we won't consolidate
+        DataFrame(self.dict_of_categoricals)
+
+
+class FromSeries:
+    def setup(self):
+        mi = MultiIndex.from_product([range(100), range(100)])
+        self.s = Series(np.random.randn(10000), index=mi)
+
+    def time_mi_series(self):
+        DataFrame(self.s)
+
+
+class FromDictwithTimestamp:
```

Comment:
We do not have a special implementation, when such a data type is used in a dictionary, it can be deleted.

Reply:
Why would the lack of a special implementation matter?

Reply:
My guess is that we don't want to have a universal test suite for pandas and modin to begin with, but rather a sufficient test suite for all modin mechanisms. So the modin mechanism for processing dictionaries does not depend on the data types inside.